src/main/java/seedu/address/logic/commands/TakeAttendanceCommand.java

```diff
@@ -0,0 +1,165 @@
+    public static final String MESSAGE_SUCCESS = "Attendance taken";
+
+    private SessionNumber sessionNumber;
+    private Name name;
+    private Set<Name> names;
+    private SessionStudents students;
+    private Session session;
+    private AttendancePresence attendancePresence;
+
+    /**
+     * Creates a `TakeAttendanceCommand` to take attendance for a single student.
+     *
+     * @param sessionNumber The session number to take attendance in.
+     * @param name The name of the student.
+     * @param attendancePresence The attendance status of the student (present/absent).
+     */
+    public TakeAttendanceCommand(SessionNumber sessionNumber, Name name, AttendancePresence attendancePresence) {
+        requireAllNonNull(sessionNumber, name, attendancePresence);
+
+        this.sessionNumber = sessionNumber;
+        this.name = name;
+        this.attendancePresence = attendancePresence;
+    }
+
+    /**
+     * Creates a `TakeAttendanceCommand` to take attendance for multiple students.
+     *
+     * @param sessionNumber The session number to take attendance in.
+     * @param names The names of the students.
+     * @param attendancePresence The attendance status of the students (present/absent).
+     */
+    public TakeAttendanceCommand(SessionNumber sessionNumber, Set<Name> names, AttendancePresence attendancePresence) {
+        requireAllNonNull(sessionNumber, names);
+
+        this.sessionNumber = sessionNumber;
+        this.names = names;
+        this.attendancePresence = attendancePresence;
+    }
+
+    /**
+     * Executes the `TakeAttendanceCommand` to take the attendance of students in the model.
+     *
+     * @param model The model to execute the command on.
+     * @return A `CommandResult` containing the success message.
+     * @throws CommandException If an error occurs during execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        this.session = model.findSessionBySessionNumber(sessionNumber);
+        this.students = new SessionStudents();
+
+        if (name != null) {
+            // Get the student to add to the session
+            Person student = model.getMatchingStudentName(name);
+            this.students.add(student);
+            if (attendancePresence.isPresent()) {
+                student.attendSession(this.session);
+            } else {
+                student.missSession(this.session);
+            }
+        }
+
+        if (names != null && !names.isEmpty()) {
+            for (Name name : names) {
+                Person student = model.getMatchingStudentName(name);
+                this.students.add(student);
+                if (attendancePresence.isPresent()) {
+                    student.attendSession(this.session);
+                } else {
+                    student.missSession(this.session);
+                }
+            }
+        }
```

Comment:
I think this code could be condensed and simplified.

Reply:
One of which checks for the nullity of "name", while the other checks for the presence of "names". Do you have any suggestions as to how this could be simplified?

I was thinking maybe we could add "name" into "names" when we instantiate the command